From 36776369d28a640e32510acff7e3deeb807a727d Mon Sep 17 00:00:00 2001
From: Mingliang Liu <liuml07@apache.org>
Date: Thu, 18 Aug 2016 00:04:58 -0700
Subject: [PATCH 2049/2848] HADOOP-13512. ReloadingX509TrustManager should
 keep reloading in case of exception. (Contributed
 by Mingliang Liu)

(cherry picked from commit 0f51eae0c085ded38216824377acf8122638c3a5)
(cherry picked from commit eae9b691ca4f9b89097d0b97d55939dc86f062a4)

Change-Id: Ib92aefd06b7b173ad55bfaaa3baf472c9d91ecba
---
 .../security/ssl/ReloadingX509TrustManager.java    |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/ssl/ReloadingX509TrustManager.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/ssl/ReloadingX509TrustManager.java
index 1b24940..3e1982e 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/ssl/ReloadingX509TrustManager.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/ssl/ReloadingX509TrustManager.java
@@ -160,10 +160,10 @@ X509TrustManager loadTrustManager()
   throws IOException, GeneralSecurityException {
     X509TrustManager trustManager = null;
     KeyStore ks = KeyStore.getInstance(type);
-    lastLoaded = file.lastModified();
     FileInputStream in = new FileInputStream(file);
     try {
       ks.load(in, password.toCharArray());
+      lastLoaded = file.lastModified();
       LOG.debug("Loaded truststore '" + file + "'");
     } finally {
       in.close();
-- 
1.7.9.5

